pkg/operator/controller/garden/components.go

```diff
@@ -521,6 +524,23 @@ func (r *Reconciler) newIstio(garden *operatorv1alpha1.Garden) (istio.Interface,
 	)
 }
 
+func (r *Reconciler) newSNI(garden *operatorv1alpha1.Garden, istioValues istio.IngressGatewayValues) component.Deployer {
+	return kubeapiserverexposure.NewSNI(
+		r.RuntimeClientSet.Client(),
+		r.RuntimeClientSet.Applier(),
+		namePrefix+v1beta1constants.DeploymentNameKubeAPIServer,
+		r.GardenNamespace,
+		func() *kubeapiserverexposure.SNIValues {
+			return &kubeapiserverexposure.SNIValues{
```

Comment:
Why don't we need the `EnvoyFilter` in the garden case?

Reply:
It's required if the Apiserver needs to be reached via the Proxy protocol. This is our approach to make `kubernetes.default.svc.cluster.local` working from within a cluster while the Apiserver is placed outside. We don't need this for the virtual cluster as there is no workload or Apiserver-Proxy DaemonSet.